plugins/aws/s3/test/unit/test_s3.py

```diff
@@ -0,0 +1,223 @@
+import io
+from dataclasses import dataclass
+from typing import Optional
+
+import pytest
+from hopeit.aws.s3 import (
+    ConnectionConfig,
+    ItemLocator,
+    ObjectStorage,
+    ObjectStorageSettings,
+)
+from hopeit.dataobjects import dataobject
+from moto.server import ThreadedMotoServer
+
+
+# Fixture to start and stop the Moto server
+@pytest.fixture(scope="session")
+def moto_server():
+    server = ThreadedMotoServer(port=9002)
+    server.start()
+    yield server
+    server.stop()
+
+
+@dataobject
+@dataclass
+class AwsMockData:
+    test: str
+
+
+@dataobject
+@dataclass
+class AwsMockEmpty:
+    test: Optional[str] = None
+
+
+test_aws = AwsMockData(test="test_aws")
+test_aws_none = AwsMockEmpty()
+
+
+@pytest.mark.asyncio
+async def test_objects_related_tasks(moto_server, monkeypatch):
+    """
+    This test verifies the behavior of object storage operations when using
+    AWS credentials from environment variables.
+    """
+    monkeypatch.setenv("AWS_ACCESS_KEY_ID", "hopeit")
+    monkeypatch.setenv("AWS_SECRET_ACCESS_KEY", "Hopei#Engine#2020")
+    monkeypatch.setenv("AWS_DEFAULT_REGION", "eu-central-1")
+
+    settings = ObjectStorageSettings(
+        bucket="test",
+        connection_config=ConnectionConfig(endpoint_url="http://localhost:9002"),
+    )
+    object_store = await ObjectStorage.with_settings(settings)
+
+    object_none = await object_store.get(key="test", datatype=AwsMockData)
+    assert object_none is None
+    empty_file = io.BytesIO(b"")
+    location = await object_store.store_file(
+        file_name="test_none.json", value=empty_file
+    )
+    assert location == "test_none.json"
+    object_none = await object_store.get(key="test_none", datatype=AwsMockEmpty)
+    assert object_none is None
+    location = await object_store.store(key="test", value=test_aws)
+    assert location == "test.json"
+    object_get = await object_store.get(key="test", datatype=AwsMockData)
+    assert object_get == test_aws
+    items = await object_store.list_objects()
+    assert items == [
```

Comment:
Is the order deterministic?

Reply:
no, it is not, I'll fix the test to avoid unexpected behavior